dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/security/AuthorisationData.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.security;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.enterprise.inject.spi.Annotated;
+import jakarta.enterprise.inject.spi.AnnotatedMethod;
+
+/**
+ *
+ */
+public class AuthorisationData {
```

Comment:
Could we make this class a `record` please? I think it would simplify it and make it clearly immutable (which it should be).